ctru-rs/src/error.rs

```diff
@@ -1,11 +1,45 @@
 use std::error;
 use std::ffi::CStr;
 use std::fmt;
+use std::ops::{ControlFlow, FromResidual, Try};
 
 use ctru_sys::result::{R_DESCRIPTION, R_LEVEL, R_MODULE, R_SUMMARY};
 
 pub type Result<T> = ::std::result::Result<T, Error>;
 
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+#[repr(transparent)]
+pub(crate) struct LibCtruResult(pub i32);
+
+impl Try for LibCtruResult {
+    type Output = ();
+    // This type is passed to [FromResidual::from_residual] when the LibCtruResult is an error,
+    // so this type implies "this is a result than CAN'T be `Ok`" (Infallible is the same as !)
+    // The purpose of this type is to _only_ bring information about the *Error*
+    type Residual = crate::Result<core::convert::Infallible>;
```

Comment:
I think this implementation can be improved by looking at this example (which describes exactly what you're trying to do here!):
https://rust-lang.github.io/rfcs/3058-try-trait-v2.html#implementing-try-for-a-non-generic-type

Reply:
Oh that's great! I'll check it out later!

Reply:
Mmm, I've not been able to create a working implementation with the example without making it more verbose or without having to use weird gymastics. I think maybe we should land this with the current impl (after I resolve the other concerns ofc), and make the implementation clearer in a follow-up PR